enhancements/insights/pulling-sca-certs-from-ocm.md

```diff
@@ -0,0 +1,166 @@
+
+### Non-Goals
+
+- Insights Operator providing any transformation or post-processing of the SCA certs pulled
+  from the OCM API
+
+## Proposal
+
+### Why is it in the Insights Operator?
+
+The Insights Operator is now the only OCP component that connects an OpenShift cluster to a Red Hat subscription experience (console.redhat.com APIs). The consumers of the SCA certs are not only builds, but also shared resources, such as the CSI driver.
+
+### User Stories
+
+#### Consume SCA certs exposed in the API
+
+As an OpenShift user
+I want to consume SCA certs to be able to consume RHEL content and to build
+corresponding container images.
+
+### Risks and Mitigations
+
+#### OCM API is down
+
+Risk: OCM API is down or doesn't provide up to date data.
+
+Risk: Insights Operator is unable to expose/update the data in the OpenShift API
+
+Mitigation: The Insights Operator is marked as Degraded (in case the HTTP code is lower than 200 or greater than 399 and is not equal to 404, because HTTP 404 means that the organization didn't allow this feature).
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the certificates from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config-managed` namespace).
+
+The Insights Operator must provide a cluster ID as an identifier of the cluster.
+
+### SCA certs in API
+
+The SCA certificate is available via the `etc-pki-entitlement` secret in the `openshift-config-managed` namespace. The secret will be available for use in other namespaces by creating a cluster-scoped `Share` resource. Cluster admin creates a `clusterrolebinding` to allow a service account access to the `Share` resource.
+
+### Use of the SCA certs
+
+- The SCA certificate can be mounted to a `Pod` as a CSI volume (where the volume attributes will reference the `Share` resource making the secret accessible)
+- The SCA certificate can be mounted to a `Build` strategy as a CSI volume. The CSI driver is described in the [Share Secrets And ConfigMaps Across Namespaces via a CSI Driver](/enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md) enhancement and implemented in [https://github.com/openshift/csi-driver-shared-resource](https://github.com/openshift/csi-driver-shared-resource).
+
+### Configuration
+- The new configuration is tied to the Insights Operator, because the Insights Operator is currently the only component reading and using the `console.redhat.com` authentication token in the `pull-secret` in the `openshift-config` namespace, so if we decide to move this somewhere else, we must consider this fact.
+- The new OCM API related config attributes are adjustable via `support` secret in the `openshift-config` namespace
+- Following attributes are configurable via the `support` secret:
+  - OCM API endpoint
+  - time period to query the OCM API
+  - flag to disable this functionality
```

Comment:
None of these three attributes sound particularly sensitive to me.  Maybe this should be a ConfigMap and not a Secret?  Or am I missing something about why one or all of them is sensitive?

Reply:
Yes, Ben is right. The `support` secret is already used for the Insights config stuff (ingress url, gathering period) and I don't know either why a secret is used as the config resource.

Reply:
Well maybe because you can configure there also username/password for the basic auth https://github.com/openshift/insights-operator/blob/master/pkg/config/configobserver/configobserver.go#L134.